File: SKIBin_old/app/src/main/java/com/h2Invent/skibin/ChildListAdapter.java
package com.h2Invent.skibin;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.android.volley.RequestQueue;
import com.google.android.material.button.MaterialButton;

import java.util.ArrayList;

public class ChildListAdapter extends RecyclerView.Adapter<ChildListAdapter.ChildListViewHolder> {
    private Context mContext;
    private ArrayList<ChildListItem> mChildList;
    private OnItemClickListener mListener;
    private RequestQueue requestQueue;

    public interface OnItemClickListener {
        void onItemClick(int position);

        void onCheckinClick(int position);
    }

    public void setmListener(OnItemClickListener listener) {
        this.mListener = listener;
    }

    public ChildListAdapter(Context context, ArrayList<ChildListItem> childList) {
        mContext = context;
        mChildList = childList;
    }

    @NonNull
    @Override
    public ChildListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(mContext).inflate(R.layout.childlistelement, parent, false);
        return new ChildListViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ChildListViewHolder holder, int position) {
        ChildListItem currentItem = mChildList.get(position);
        String name = currentItem.getmName();
        String schule = currentItem.getmSchule();
        int klasse = currentItem.getmKlasse();
        int schulId = currentItem.getmSchulId();
        boolean checkin = currentItem.ismCheckin();
        boolean hasBirthday = currentItem.isMhasBirthday();
        holder.textViewname.setText(name);
        holder.textViewschule.setText(schule);

        if (checkin && schulId > -1) {
            holder.linearLayoutIndicator.setBackgroundColor(ContextCompat.getColor(mContext, R.color.backgroundSuccess));
            holder.mCheckinBtn.setVisibility(View.GONE);
            holder.mCheckinBtn.setEnabled(false);
        } else if (!checkin && schulId > -1)  {
            holder.linearLayoutIndicator.setBackgroundColor(ContextCompat.getColor(mContext, R.color.backgroundError));
        }
        if(schulId == -1){
            holder.mCheckinBtn.setVisibility(View.GONE);
            holder.mCheckinBtn.setEnabled(false);
        }
        if (klasse > 0) {
            holder.textViewklasse.setText(String.valueOf(klasse) + ". Klasse");
        } else {
            holder.textViewklasse.setText("");
        }
        if (hasBirthday) {
            holder.hasBirthday.setVisibility(View.VISIBLE);
        }

    }

    @Override
    public int getItemCount() {
        return mChildList.size();
    }

    public class ChildListViewHolder extends RecyclerView.ViewHolder {
        public TextView textViewname;
        public TextView textViewschule;
        public TextView textViewklasse;
        public LinearLayout linearLayoutIndicator;
        public TextView hasBirthday;
        public MaterialButton mCheckinBtn;

        public ChildListViewHolder(@NonNull View itemView) {
            super(itemView);
            textViewname = (TextView) itemView.findViewById(R.id.childElementName);
            textViewschule = (TextView) itemView.findViewById(R.id.childElementSchule);
            textViewklasse = (TextView) itemView.findViewById(R.id.childElementKlasse);
            linearLayoutIndicator = (LinearLayout) itemView.findViewById(R.id.indicatorCheckin);
            hasBirthday = (TextView) itemView.findViewById(R.id.birthdayShow);
            mCheckinBtn = (MaterialButton) itemView.findViewById(R.id.checkinButton);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (mListener != null) {
                        int position = getAbsoluteAdapterPosition();
                        if (position != RecyclerView.NO_POSITION) {
                            mListener.onItemClick(position);
                        }
                    }
                }
            });
            mCheckinBtn.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (mListener != null) {
                        int position = getAbsoluteAdapterPosition();
                        if (position != RecyclerView.NO_POSITION) {
                            mListener.onCheckinClick(position);
                        }
                    }
                }
            });


        }
    }
}

File: SKIBin_old/app/src/main/java/com/h2Invent/skibin/ChildListAdapter.kt
package com.h2Invent.skibin

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.button.MaterialButton

class ChildListAdapter(
    private val items: List<ChildListItem>,
    private val listener: OnItemClickListener,
) : RecyclerView.Adapter<ChildListAdapter.ChildListViewHolder>() {

    interface OnItemClickListener {
        fun onItemClick(position: Int)
        fun onCheckinClick(position: Int)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChildListViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.childlistelement, parent, false)
        return ChildListViewHolder(view)
    }

    override fun onBindViewHolder(holder: ChildListViewHolder, position: Int) {
        val item = items[position]
        holder.textViewName.text = item.name
        holder.textViewSchool.text = item.school
        holder.textViewGrade.text = if (item.grade > 0) "${item.grade}. Klasse" else ""
        holder.hasBirthday.visibility = if (item.hasBirthday) View.VISIBLE else View.GONE

        when {
            item.schoolId == -1 -> {
                holder.indicator.setBackgroundColor(ContextCompat.getColor(holder.itemView.context, android.R.color.transparent))
                holder.checkinButton.visibility = View.GONE
                holder.checkinButton.isEnabled = false
            }
            item.checkedIn -> {
                holder.indicator.setBackgroundColor(ContextCompat.getColor(holder.itemView.context, R.color.backgroundSuccess))
                holder.checkinButton.visibility = View.GONE
                holder.checkinButton.isEnabled = false
            }
            else -> {
                holder.indicator.setBackgroundColor(ContextCompat.getColor(holder.itemView.context, R.color.backgroundError))
                holder.checkinButton.visibility = View.VISIBLE
                holder.checkinButton.isEnabled = true
            }
        }
    }

    override fun getItemCount(): Int = items.size

    inner class ChildListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val textViewName: TextView = itemView.findViewById(R.id.childElementName)
        val textViewSchool: TextView = itemView.findViewById(R.id.childElementSchule)
        val textViewGrade: TextView = itemView.findViewById(R.id.childElementKlasse)
        val indicator: LinearLayout = itemView.findViewById(R.id.indicatorCheckin)
        val hasBirthday: TextView = itemView.findViewById(R.id.birthdayShow)
        val checkinButton: MaterialButton = itemView.findViewById(R.id.checkinButton)

        init {
            itemView.setOnClickListener {
                val position = bindingAdapterPosition
                if (position != RecyclerView.NO_POSITION) listener.onItemClick(position)
            }
            checkinButton.setOnClickListener {
                val position = bindingAdapterPosition
                if (position != RecyclerView.NO_POSITION) listener.onCheckinClick(position)
            }
        }
    }
}
